src/shims/time.rs

```diff
@@ -159,4 +160,25 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         u64::try_from(duration.as_nanos())
             .map_err(|_| err_unsup_format!("programs running longer than 2^64 nanoseconds are not supported").into())
     }
+
+    fn mach_timebase_info(&mut self, info_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        this.assert_target_os("macos", "mach_timebase_info");
+        this.check_no_isolation("mach_timebase_info")?;
+
+        let info = this.deref_operand(info_op)?;
+
+        // Since we return nanoseceonds instead of ticks from
+        // `mach_absolute_time`, we don't need to scale the absolute
+        // time.
+        let (numer, denom) = (1,1);
+        let imms = [
+            immty_from_int_checked(numer, this.libc_ty_layout("uint32_t")?)?,
```

Comment:
`u32` layout is available in `machine.layouts`, please use that instead of indirecting through libc.